system/ee/ExpressionEngine/Addons/channel/mod.channel.php

```diff
@@ -3951,7 +3951,7 @@ public function close_ul($parent_id, $depth = 0)
       */
     public function category_heading()
     {
-        if ($this->query_string == '') {
+        if ($this->query_string === '' && !ee()->TMPL->fetch_param('category_url_title') && !ee()->TMPL->fetch_param('category_id')) {
```

Comment:
Just out of curiosity, has there been specific reason to also check for type (`===`) of `$this->query_string`. I think it's always string, so should be safe to do, so just wondering